### PR TITLE
Fix for pending batch txs mismatch

### PR DESCRIPTION
**Pull request by iulianpascalau**
- fix for pending batch that might get different number of transactions when setting the batch transfer status
- added more prints

### CODE REVIEW

**Review by sasurobert — APPROVED** [review]
Still, both contracts should give only final data, which cannot be forked when asking for getPendingBatch

**Review by sasurobert — APPROVED** [review]
Still, both contracts should give only final data, which cannot be forked when asking for getPendingBatch

**Review by dragos-rebegea — DISMISSED** [review]

**Review by dragos-rebegea — APPROVED** [review]

**Review by dragos-rebegea — DISMISSED** [review]

File: bridge/elrond/client.go [hunk 1]
```diff
@@ -189,13 +190,21 @@ func parseIntFromByteSlice(buff []byte) (int64, error) {
...
+func (c *client) ProposeSetStatus(ctx context.Context, batch *bridge.Batch) {
 	builder := newBuilder(c.log).
 		Func("proposeEsdtSafeSetCurrentTransactionBatchStatus").
 		BatchId(batch.Id)
 
-	for _, tx := range batch.Transactions {
-		builder = builder.Int(big.NewInt(int64(tx.Status)))
+	newBatch, err := c.GetPending(ctx)
+	if err != nil {
+		c.log.Error("Elrond: get pending batch failed in ProposeSetStatus", "error", err.Error())
```
Comment (bogdan-rosianu) [inline bridge/elrond/client.go:200]: optional: could use `err` instead of `err.Error()`

File: bridge/elrond/client.go [hunk 2]
```diff
@@ -220,13 +229,17 @@ func (c *client) ProposeTransfer(_ context.Context, batch *bridge.Batch) (string
...
+
 	gasLimit := c.gasMapConfig.ProposeTransferBase + uint64(len(batch.Transactions))*c.gasMapConfig.ProposeTransferForEach
 	hash, err := c.sendTransaction(builder, gasLimit)
-
 	if err == nil {
-		c.log.Info("Elrond: Proposed transfer for batch ", batch.Id, " with hash ", hash)
+		c.log.Info("Elrond: Proposed transfer for batch ", batch.Id, "with hash", hash, "batch data", string(batchData))
 	} else {
-		c.log.Error("Elrond: Propose transfer errored", "error", err.Error())
+		c.log.Error("Elrond: Propose transfer errored", "batch data", string(batchData), "error", err.Error())
```
Comment (bogdan-rosianu) [inline bridge/elrond/client.go:242]: optional: `err` instead of `err.Error()`

File: bridge/elrond/client.go [hunk 3]
```diff
@@ -265,13 +278,20 @@ func (c *client) GetActionIdForProposeTransfer(_ context.Context, batch *bridge.
...
+func (c *client) WasProposedSetStatus(ctx context.Context, batch *bridge.Batch) bool {
 	valueRequest := newValueBuilder(c.bridgeAddress, c.address.AddressAsBech32String(), c.log).
 		Func("wasSetCurrentTransactionBatchStatusActionProposed").
 		BatchId(batch.Id)
 
-	for _, tx := range batch.Transactions {
-		valueRequest = valueRequest.BigInt(big.NewInt(int64(tx.Status)))
+	newBatch, err := c.GetPending(ctx)
+	if err != nil {
+		c.log.Error("Elrond: get pending batch failed in WasProposedSetStatus", "error", err.Error())
```
Comment (bogdan-rosianu) [inline bridge/elrond/client.go:288]: optional: `err` instead of `err.Error()`

File: bridge/elrond/client.go [hunk 4]
```diff
@@ -324,13 +346,20 @@ func getStatusFromBuff(buff []byte) (byte, error) {
...
+func (c *client) GetActionIdForSetStatusOnPendingTransfer(ctx context.Context, batch *bridge.Batch) bridge.ActionId {
 	valueRequest := newValueBuilder(c.bridgeAddress, c.address.AddressAsBech32String(), c.log).
 		Func("getActionIdForSetCurrentTransactionBatchStatus").
 		BatchId(batch.Id)
 
-	for _, tx := range batch.Transactions {
-		valueRequest = valueRequest.BigInt(big.NewInt(int64(tx.Status)))
+	newBatch, err := c.GetPending(ctx)
+	if err != nil {
+		c.log.Error("Elrond: get pending batch failed in WasProposedSetStatus", "error", err.Error())
```
Comment (bogdan-rosianu) [inline bridge/elrond/client.go:356]: optional: `err` instead of `err.Error()`

File: bridge/elrond/client.go [hunk 5]
```diff
@@ -359,17 +390,22 @@ func (c *client) WasExecuted(_ context.Context, actionId bridge.ActionId, _ brid
...
+	if err != nil {
+		c.log.Warn("Elrond: error not critical while serializing transaction", "error", err)
+	}
+
 	if err == nil {
-		c.log.Info("Elrond: Signed", "hash", hash)
+		c.log.Info("Elrond: Signed", "hash", hash, "batch data", string(batchData))
 	} else {
-		c.log.Error("Elrond: Sign failed", "error", err.Error())
+		c.log.Error("Elrond: Sign failed", "batch data", string(batchData), "error", err.Error())
```
Comment (bogdan-rosianu) [inline bridge/elrond/client.go:408]: optional: `err` instead of `err.Error()`

File: bridge/elrond/client.go [hunk 6]
```diff
@@ -384,17 +420,26 @@ func (c *client) Execute(_ context.Context, actionId bridge.ActionId, batch *bri
...
 	if err == nil {
-		c.log.Info("Elrond: Executed action", "action ID", actionId, "hash", hash)
+		c.log.Info("Elrond: Executed action", "actionID", actionId, "batch data", string(batchData), "hash", hash)
 	} else {
-		c.log.Info("Elrond: Execution failed for action", "action ID", actionId, "hash", hash, "error", err.Error())
+		c.log.Info("Elrond: Execution failed for action",
+			"actionID", actionId,
+			"batch data", string(batchData),
+			"hash", hash,
+			"error", err.Error())
```
Comment (bogdan-rosianu) [inline bridge/elrond/client.go:435]: optional: `err` instead of `err.Error()`

File: bridge/models.go [hunk 7]
```diff
@@ -3,10 +3,16 @@ package bridge
 import (
 	"fmt"
 	"math/big"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
 )
 
+var log = logger.GetOrCreate("bridgemodels")
```
Comment (bogdan-rosianu) [inline bridge/models.go:10]: optional: `bridge/models`

File: bridge/models.go [hunk 8]
```diff
@@ -34,24 +41,49 @@ type DepositTransaction struct {
...
+
+	for i := 0; i < len(batch.Statuses); i++ {
+		batch.Statuses[i] = status
+	}
+}
+
+// ResolveNewDeposits will add new statuses as rejected if the newNumDeposits exceeds the number of the deposits
+func (batch *Batch) ResolveNewDeposits(newNumDeposits int) {
+	oldLen := len(batch.Statuses)
+	if newNumDeposits == len(batch.Statuses) {
```
Comment (bogdan-rosianu) [inline bridge/models.go:74]: newNumDeposits == oldLen

File: bridge/models.go [hunk 9]
```diff
@@ -34,24 +41,49 @@ type DepositTransaction struct {
...
+		log.Debug("num statuses ok", "len statuses", oldLen)
+		return
+	}
+
+	if newNumDeposits < len(batch.Statuses) {
```
Comment (bogdan-rosianu) [inline bridge/models.go:79]: oldLen instead of len(..)

**Review by mihaieremia — APPROVED** [review]

File: bridge/elrond/client.go (hunk 1 above)
Comment (iulianpascalau) [reply bridge/elrond/client.go:200]: done

File: bridge/elrond/client.go (hunk 2 above)
Comment (iulianpascalau) [reply bridge/elrond/client.go:242]: done

File: bridge/elrond/client.go (hunk 3 above)
Comment (iulianpascalau) [reply bridge/elrond/client.go:288]: done

File: bridge/elrond/client.go (hunk 4 above)
Comment (iulianpascalau) [reply bridge/elrond/client.go:356]: done

File: bridge/elrond/client.go (hunk 5 above)
Comment (iulianpascalau) [reply bridge/elrond/client.go:408]: done

File: bridge/elrond/client.go (hunk 6 above)
Comment (iulianpascalau) [reply bridge/elrond/client.go:435]: done

File: bridge/models.go (hunk 7 above)
Comment (iulianpascalau) [reply bridge/models.go:10]: done

File: bridge/models.go (hunk 8 above)
Comment (iulianpascalau) [reply bridge/models.go:74]: done

File: bridge/models.go (hunk 9 above)
Comment (iulianpascalau) [reply bridge/models.go:79]: done

**Review by bogdan-rosianu — APPROVED** [review]

**Review by dragos-rebegea — APPROVED** [review]

**Review by sasurobert — COMMENTED** [review]
Approve. Nicely done

**Pull request merged**